src/components/PermissionsManagement/UserPermissionsPopUp.jsx

```diff
@@ -80,9 +90,40 @@ function UserPermissionsPopUp({
     const userId = actualUserProfile?._id;
 
     const url = ENDPOINTS.USER_PROFILE(userId);
+    const logPermissionChangesUrl = ENDPOINTS.POST_USER_PERMISSION_CHANGE_LOGS;
     const allUserInfo = await axios.get(url).then(res => res.data);
     const newUserInfo = { ...allUserInfo, permissions: { frontPermissions: userPermissions } };
 
+    let existingPermissions = allUserInfo.permissions?.frontPermissions || [];
+    const existingPermissionsSet = new Set(existingPermissions || []);
+    const newPermissionsSet = new Set(userPermissions || []);
+    const addedPermissions = (userPermissions || []).filter(
+      permission => !existingPermissionsSet.has(permission),
+    );
+    const removedPermissions = (existingPermissions || []).filter(
+      permission => !newPermissionsSet.has(permission),
+    );
+
+    existingPermissions = [
+      ...existingPermissions.filter(permission => !removedPermissions.includes(permission)),
+      ...addedPermissions,
+    ];
+
+    const permissionChangeLog = {
+      actualUserProfile: {
+        firstName: actualUserProfile.firstName,
+        lastName: actualUserProfile.lastName,
+      },
+      authUser: {
+        email: authUser.email,
+        role: authUser.role,
+      },
+      userId,
+      existingPermissions, // Existing permissions before change
+      addedPermissions, // New permissions added
+      removedPermissions, // Permissions removed
+    };
+
```

Comment:
```suggestion
```
This logic should be on the backend and part of the same API call as the update itself.